### DISCUSSION TITLE
Can this be improved further ? 4090 Performance Tweaking

### DISCUSSION THREAD

**Discussion by ameen-roayan** [vladmandic/sdnext#699] [Q&A]
Hello ! 
I recently moved away from 3090ti to an Asus 4090, the results thus far have not been that great and I am slowly trying to understand what is going on. the results are not far from each other, and I am reading much better results from others and did some of the tweaks but nothing changed much. 

![image](https://user-images.githubusercontent.com/92278134/235678139-02a26f63-bbb7-49f6-a229-359770717303.png)
![image](https://user-images.githubusercontent.com/92278134/235678207-ed75c349-d2fe-4b23-8904-1f7c7157d142.png)

current generation 512x512 is at around ~20 it/s 

![image](https://user-images.githubusercontent.com/92278134/235678467-0aec4089-be2a-4ae2-8bea-c1cab6ebadbb.png)

Appreciate the feedback

**Comment by vladmandic** [comment]
I'm afraid that Coffee Lake CPU may not have enough bandwidth to fully saturate RTX4090 GPU, especially since that CPU+MB only have PCIe 3.0, not 4.0 - data just cannot come to GPU fast enough.
Since you're on Windows 11, make sure that GPU scheduling is enabled, other than that, nothing obvious comes to mind.
(Settings > System > Display > Graphics > Change Default Graphics Settings > Hardware-Accelerated GPU Scheduling)

**Reply by BasedAnon** [reply]
So, GPU scheduling HELPS Stable Diffusion?
Everyone else told me to turn it off!  (I have since switched to Linux for all my Python use)

**Reply by vladmandic** [reply]
It depends. High end CPU and you want to disable it to keep GPU as free as possible. High end GPU with mid-level CPU and you want to enable it as that CPU cannot saturate GPU enough. Its all about getting the most out of the combo.

**Reply by Tillerz** [reply]
Are all the real fast results on Linux maybe, where the models get compiled or something? Even the ones that they they are Windows machines? I don't know which part is doing the reporting, but if you have a windows pc and the servers is running inside a virtual box on linux, it might report as a windows machine?

**Reply by BasedAnon** [reply]
> It depends. High end CPU and you want to disable it to keep GPU as free as possible. High end GPU with mid-level CPU and you want to enable it as that CPU cannot saturate GPU enough. Its all about getting the most out of the combo.

I've got a 4090 GPU and an i9-11900K:  is that good enough to avoid a CPU-GPU bottleneck?
(If not, then wow... :skull: )

(also, thanks for the timely and informative reply!)

**Reply by vladmandic** [reply]
i9-119000k? best guess probably not but really close. i'd say 12900k would be sufficient. and yeah, its wow how hungry is 4090.
btw, just to see what cpu bottlneck does, see benchmarks in system-info extension - first number is batch-size 1 so cpu needs to schedule new job as soon as first one is done, etc. and last is batch 4 - its nearly double performance since amount of cpu scheduling is about the same and gpu does 4x work.

**Reply by Tillerz** [reply]
I just tried with both (on and off for GPU scheduling) and it seemed it was quite some slower with scheduling OFF for the larger batches (AMD Ryzen 7 7700X 8-Core Processor). I have to fiddle around and check more though. Because I don't do batches but I render 1 image and then fiddle with the prompt or the sliders, I don't need big performance on batches, just for the 1 render with upscale to 1024x10124 or 1280x1280.

**Reply by YukiSakuma** [reply]
Can someone explain to me why CPU still affects the generation time? I thought it's just the GPU doing the whole work.

**Reply by vladmandic** [reply]
oversimplified: gpu is doing all of the actual calculations, but something needs to tell gpu what to do.
